frame/staking/src/pallet/mod.rs

```diff
@@ -1505,7 +1507,7 @@ pub mod pallet {
 			});
 
 			// NOTE: ledger must be updated prior to calling `Self::weight_of`.
-			Self::update_ledger(&controller, &ledger);
+			ledger.update()?;
```

Comment:
I find this comment a bit confusing, I don't see `Self::weight_of` called prior to this

Reply:
The call to `Self::weight_of` is made after the update, so that the `VoterList` is updated with the correct, up to date, ledger state. I hope that with the upcoming PR that refactors stake state updates, the ordering of these operations will be better documented and explicit.